### ISSUE TITLE
Bug: Process crashes when running the query

### ISSUE THREAD

**Issue by andriichumak** [kuzudb/kuzu#3593]
### Kùzu version

v0.4.2

### What happened?

The process is crashing without any error message when I try to execute the query. Tried in my Python app and in Kuzu Explorer - same result. I suspect this is due to circular MATCH, which was my mistake, but I'd expect to see some error message in this case.

### Are there known steps to reproduce?

```
MATCH (n:Workspace {id: $workspace_id})-[r*0..]-(i)
DETACH DELETE n, i
```
Fixed it by adding connection direction and limited the number of links to 10, just in case.
```
MATCH (n:Workspace {id: $workspace_id})-[r*0..10]->(i)
DETACH DELETE n, i
```
`RETURN` instead of `DETACH DELETE` yields the same result.

**Comment by andyfengHKU** [comment]
Just looked into this issue. It seems we using exceptional large amount of memory while doing this DETACH DELETE. And an exception was thrown when we try to free used memory in `BufferManager::freeUsedMemory`

This is reproducible with tinysnb dataset with query
```
MATCH (a:person)-[e*0..10]->(b:person) DETACH DELETE a, b;
```

I'm forwarding this issue to @ray6080.

**Comment by andyfengHKU** [comment]
Hi @andriichumak this should be solved in #3637
